modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java

```diff
@@ -3051,6 +3073,37 @@ else if (tx.setRollbackOnly())
                 cctx.kernalContext().gateway().readUnlock();
             }
         }
+
+        /**
+         * @param tx Tx.
+         * @param failedNode Failed node.
+         */
+        private void processPrepared(IgniteInternalTx tx, UUID failedNode) {
+            IgniteInternalFuture<Boolean> fut = commitIfPrepared(tx, Collections.singleton(failedNode));
+
+            if (fut != null)
+                doneFut.add(fut);
+
+            txCnt.incrementAndGet();
+        }
+
+        /**
+         *
+         */
+        private void finishAndRecordTimings() {
+            timeBag.finishGlobalStage("Finished");
+
+            StringBuilder timingsToLog = new StringBuilder();
+
+            timingsToLog.append("TxRecovery Status and Timings [txs=").append(txCnt.get());
+
+            for (String stageTiming : timeBag.stagesTimings())
+                timingsToLog.append(", ").append(stageTiming);
+
+            timingsToLog.append("]");
+
+            log.info(timingsToLog.toString());
```

Comment:
Let's wrap in:

```
if (log.isInfoEnabled()) {
            StringBuilder timingsToLog = new StringBuilder();

            timingsToLog.append("TxRecovery Status and Timings [txs=").append(txCnt.get());

            for (String stageTiming : timeBag.stagesTimings())
                timingsToLog.append(", ").append(stageTiming);

            timingsToLog.append("]");

            log.info(timingsToLog.toString());
}
```

Reply:
do we really need to check isInfoEnabled?

Reply:
done